nemo/src/io/formats/sparql.rs

```diff
@@ -0,0 +1,186 @@
+
+format_tag! {
+    pub(crate) enum SparqlTag(SupportedFormatTag::Sparql) {
+        Sparql => file_format::SPARQL,
+    }
+}
+
+format_parameter! {
+    pub(crate) enum SparqlParameter(StandardParameter) {
+        Limit(name = attribute::LIMIT, supported_types = &[ValueType::Number]),
+        Format(name = attribute::FORMAT, supported_types = &[ValueType::Constant, ValueType::Tuple]),
+        Base(name = attribute::BASE, supported_types = &[ValueType::Constant]),
+        Endpoint(name = attribute::ENDPOINT, supported_types = &[ValueType::Constant]),
+        Query(name = attribute::QUERY, supported_types = &[ValueType::String]),
+        IgnoreHeaders(name = attribute::IGNORE_HEADERS, supported_types = &[ValueType::Boolean]),
+
+    }
+}
+
+impl FormatParameter<SparqlTag> for SparqlParameter {
+    fn required_for(&self, tag: SparqlTag) -> bool {
+        if matches!(tag, SparqlTag::Sparql) {
+            matches!(self, SparqlParameter::Endpoint) && matches!(self, SparqlParameter::Query)
+        } else {
+            false
+        }
+    }
+
+    fn is_value_valid(&self, value: AnyDataValue) -> Result<(), ValidationErrorKind> {
+        value_type_matches(self, &value, self.supported_types())?;
+
+        match self {
+            SparqlParameter::BaseParamType(base) => {
+                FormatParameter::<SparqlTag>::is_value_valid(base, value)
+            }
+            SparqlParameter::Limit => value
+                .to_u64()
+                .and(Some(()))
+                .ok_or(ValidationErrorKind::ImportExportLimitNegative),
+            SparqlParameter::Base => Ok(()),
+            SparqlParameter::Format => DsvValueFormats::try_from(value).and(Ok(())).map_err(|_| {
+                ValidationErrorKind::ImportExportValueFormat {
+                    file_format: "rdf".to_string(),
+                }
+            }),
+            SparqlParameter::IgnoreHeaders => Ok(()),
+            SparqlParameter::Endpoint => value
+                .to_iri()
+                .ok_or(ValidationErrorKind::ImportExportInvalidIri)
+                .and_then(|iri| {
+                    Iri::parse(iri)
+                        .map(|_| ())
+                        .map_err(|_| ValidationErrorKind::ImportExportInvalidIri)
+                }),
+            SparqlParameter::Query => value
+                .to_plain_string()
+                .ok_or(ValidationErrorKind::ImportExportLimitNegative)
+                .and_then(|query| {
+                    Query::parse(query.as_str(), None).map(|_| ()).map_err(|e| {
+                        ValidationErrorKind::ImportExportInvalidSparqlQuery {
+                            oxi_error: e.to_string(),
+                        }
+                    })
+                }),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct SparqlBuilder {
+    limit: Option<u64>,
+    delimiter: u8,
+    value_formats: Option<DsvValueFormats>,
+    ignore_headers: bool,
```

Comment:
this also shouldn't be an option, for SPARQL, we *always* want to ignore the heades